newlib/libc/tinystdio/fdopen.c

```diff
@@ -59,6 +59,7 @@ fdopen(int fd, const char *mode)
         *bf = (struct __file_bufio)
                 FDEV_SETUP_POSIX(fd, buf, BUFSIZ, stdio_flags, __BFALL);
 
+        __flockfile_init(&(bf->xfile.cfile.file));
         __bufio_lock_init(&(bf->xfile.cfile.file));
```

Comment:
We don't need a separate lock for bufio; the global file lock will suffice.

Reply:
@keith-packard , please see [this](https://github.com/Lapshin/picolibc/blob/feature/add-flockfile-build-option/newlib/libc/tinystdio/stdio_private.h#L257) and [this](https://github.com/Lapshin/picolibc/blob/feature/add-flockfile-build-option/newlib/libc/tinystdio/stdio-bufio.h#L128). Do you want me implement this in a different way?

Reply:
ah, clever. Yes, that looks great. Sorry for not reading the patch in enough detail to understand how this all works.